src/components/App.js

```diff
@@ -1,19 +1,22 @@
 import { Route, Routes } from 'react-router-dom';
-
+import { Provider } from 'react-redux';
 import './App.css';
 import Header from './Header';
 import Categories from './Categories';
 import Books from './Books';
+import store from '../redux/store';
 
 const App = () => (
   <>
-    <Header />
-    <div className="app-container">
-      <Routes>
-        <Route path="/" element={<Books />} />
-        <Route path="/categories" element={<Categories />} />
-      </Routes>
-    </div>
+    <Provider store={store}>
+      <Header />
+      <div className="app-container">
+        <Routes>
+          <Route path="/" element={<Books />} />
+          <Route path="/categories" element={<Categories />} />
+        </Routes>
+      </div>
+    </Provider>
```

Comment:
- [ OPTIONAL ] Great job @Tshobohwa 💪 Your project works as expected 💯 I will only suggest that instead of putting the `provider` here you import in in the `index.js` file together with the `redux store` and ensure that the entire app component is wrapped around it and not just some parts of it 😉 

- EG. In` index.js`

```
import React from "react"
import ReactDOM from "react-dom/client"
import { Provider } from "react-redux"
import store from "./redux/store"

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById("root")
)

```